tools/pythonpkg/src/python_objects.cpp

```diff
@@ -118,10 +120,12 @@ void PyDecimal::SetExponent(py::handle &exponent) {
 	ExponentNotRecognized();
 }
 
-static void UnsupportedWidth(uint16_t width) {
-	throw ConversionException(
-	    "Failed to convert to a DECIMAL value with a width of %d because it exceeds the max supported with of %d",
-	    width, Decimal::MAX_WIDTH_INT128);
+static bool WidthFitsInDecimal(int32_t width) {
+	return width >= 0 && width <= Decimal::MAX_WIDTH_DECIMAL;
+}
+
+static void DoubleConversionFail(const string &decimal) {
+	throw ConversionException("Failed to fall back to a DOUBLE for DECIMAL '%s'", decimal);
```

Comment:
It seems that this error is never triggered by tests, can we have a test that triggers it?

Reply:
Ah yes, weird that the compiler doesn't complain about this unused static function.
I looked into the code this is calling, and experimented with it a bunch to make it fail, but I couldn't get it to fail, so I have changed the error check into an assertion. I'll remove this function